core/src/main/java/dk/alexandra/fresco/framework/builder/numeric/field/FieldDefinition.java

```diff
@@ -0,0 +1,80 @@
+package dk.alexandra.fresco.framework.builder.numeric.field;
+
+import dk.alexandra.fresco.framework.network.serializers.ByteSerializer;
+import dk.alexandra.fresco.framework.util.StrictBitVector;
+import java.math.BigInteger;
+
+/**
+ * A field describes the algebraic structure being a finite Galois field.
+ * Every element in the field is a positive number less than the modulus.
+ */
+public interface FieldDefinition extends ByteSerializer<FieldElement> {
+
+  /**
+   * Creates an element in the field from the supplied value.
+   *
+   * @param value the value to convert into the field
+   * @return the created field element
+   */
+  FieldElement createElement(long value);
+
+  /**
+   * Creates an element in the field from the supplied value.
+   *
+   * @param value the value to convert into the field
+   * @return the created field element
+   */
+  FieldElement createElement(String value);
+
+  /**
+   * Creates an element in the field from the supplied value.
+   *
+   * @param value the value to convert into the field
+   * @return the created field element
+   */
+  FieldElement createElement(BigInteger value);
+
+  /**
+   * Gets the modulus for this finite field.
+   *
+   * @return the created field element
+   */
+  BigInteger getModulus();
+
+  /**
+   * Gets the bit length for numbers in this field.
+   *
+   * @return the bit length
+   */
+  int getBitLength();
+
+  /**
+   * Creates a StrictBitVector from an element.
+   *
+   * @param fieldElement The element to convert
+   * @return the element represented as a bit vector
+   */
+  StrictBitVector convertToBitVector(FieldElement fieldElement);
```

Comment:
Serialization is a core aspect of defining a field, since the internal values can be precisely serialized using the parameters of the field. I would like to defer considering refactoring to a later point